.github/workflows/codeql.yml

```diff
@@ -0,0 +1,43 @@
+name: "CodeQL"
+
+on:
+  push:
+    branches: [ "main" ]
+  pull_request:
+    # The branches below must be a subset of the branches above
+    branches: [ "main" ]
+  schedule:
+    # run every week at 10:24 on Thursday
+    - cron: '24 10 * * 4'
+
+jobs:
+  analyze:
+    name: Analyze
+    runs-on: ubuntu-latest
+    permissions:
+      actions: read
+      contents: read
+      security-events: write
+
+    strategy:
+      fail-fast: false
+      matrix:
+        language: [ 'go', 'javascript' ]
+
+    steps:
+    - name: Checkout repository
+      uses: actions/checkout@v3
+
+    - name: Initialize CodeQL
+      uses: github/codeql-action/init@v2
+      with:
+        languages: ${{ matrix.language }}
+
+    - name: Download Go Dependencies
+      if: matrix.language == 'go'
+      run: go mod download
```

Comment:
Based on output from https://github.com/coder/coder/actions/runs/3737608156/jobs/6342956567, it looks like Go isn't available? Go mod download results in no output. The [Python example](https://docs.github.com/en/code-security/code-scanning/automatically-scanning-your-code-for-vulnerabilities-and-errors/configuring-code-scanning#analyzing-python-dependencies) does manual setup of Python, so maybe it will help here.

Also, it looks like "Perform Analysis" tries to build a bunch of stuff, including the node/site package and takes a whole 15 minutes to complete. Maybe mod download failure is the reason, or we need another way to disable the autobuilding.

```suggestion
    - name: Setup Go
      if: matrix.language == 'go'
      uses: actions/setup-go@v3
      with:
        go-version: "~1.19"

    - name: Download Go Dependencies
      if: matrix.language == 'go'
      run: go mod download
```

Reply:
Hmm, looks like this didn't change the analysis step. Maybe the step works, but analysis still triggers a `go mod download` via build-step as well. I'm confused as to why it's behaving that way.